cmd/internal/cli/sign.go

```diff
@@ -104,17 +103,29 @@ var SignCmd = &cobra.Command{
 }
 
 func doSignCmd(cpath, url string) error {
-	if sifGroupID != 0 && sifDescID != 0 {
-		return fmt.Errorf("only one of -i or -g may be set")
+	if sifGroupID != -1 && sifDescID != -1 {
+		sylog.Fatalf("only one of -i or -g may be set")
+	}
+
```

Comment:
you could use cobra by passing `cmd *cobra.Command` to `doSignCmd` and have something like

```
if cmd.Flag(signSifGroupIDFlag.Name).Changed() && sifGroupID == 0 {
    sylog.Fatalf("invalid group id")
}
if cmd.Flag(signSifDescSifIDFlag.Name).Changed() && sifDescID == 0 {
    sylog.Fatalf("invalid descriptor id")
}
if cmd.Flag(signSifDescIDFlag.Name).Changed() && sifDescID == 0 {
    sylog.Fatalf("invalid descriptor id")
}
```